### PR TITLE
rsyslog 8.34.0

### PR DESCRIPTION
**Pull request by dirker**
remove json-c dependency
vendor libfastjson and liblogging-stdlog
add patch for https://github.com/rsyslog/rsyslog/issues/2697

Updating this also resolves compile and linkage problems with the updated json-c from https://github.com/Homebrew/homebrew-core/pull/27463.

-----

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

**Comment by dirker** [comment]
@ilovezfs: there currently is no test block in this formula, and I'm not sure I can add one since I am not familiar with rsyslog.

File: Formula/rsyslog.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class Rsyslog < Formula
   desc "Enhanced, multi-threaded syslogd"
   homepage "https://www.rsyslog.com/"
-  url "https://www.rsyslog.com/files/download/rsyslog/rsyslog-7.4.5.tar.gz"
-  sha256 "f5e46e9324e366f20368162b4f561cf7a76fecb4aa0570edcaaa49e9f8c2fe70"
+  url "http://www.rsyslog.com/files/download/rsyslog/rsyslog-8.34.0.tar.gz"
```
Comment (SMillerDev) [inline Formula/rsyslog.rb]: What's wrong with the https link?

Comment (dirker) [reply Formula/rsyslog.rb]: Oversight on my part, fixed it in latest push.

**Comment by ilovezfs** [comment]
Thanks again @dirker! Great work :)

**Pull request closed without merging**